src/Fixer/Operator/BinaryOperatorSpacesFixer.php

```diff
@@ -924,6 +924,25 @@ private function replacePlaceholders(Tokens $tokens, string $alignStrategy, stri
                     $delta = abs($rightmostSymbol - $currentSymbol);
 
                     if ($delta > 0) {
+                        if (
+                            self::ALIGN !== $alignStrategy
+                            && self::ALIGN_BY_SCOPE !== $alignStrategy
+                        ) {
+                            foreach (array_keys($this->alignOperatorTokens) as $alignOperatorToken) {
+                                $escapedAlignOperator = preg_quote($alignOperatorToken, '/');
+                                $pattern = '/(?<!'
+                                    .preg_quote($placeholder, '/')
+                                    .preg_quote($tokenContent, '/')
+                                    .')\s+('
+                                    .$escapedAlignOperator
+                                    .')/';
+
+                                if (Preg::match($pattern, $line, $matches)) {
```

Comment:
Currently `Preg::match()` is executed multiple times, for each `$this->alignOperatorTokens`. Would it be possible to call it once, with operators as an `OR` group? I've tried quick change like:

```php
$escapedAlignOperators = [];
foreach (array_keys($this->alignOperatorTokens) as $alignOperatorToken) {
    $escapedAlignOperators[] = preg_quote($alignOperatorToken, '/');
}

$pattern = '/(?<!'
    . preg_quote($placeholder, '/')
    . preg_quote($tokenContent, '/')
    . ')\s+('
    . implode('|', $escapedAlignOperators)
    . ')/';

if (Preg::match($pattern, $line, $matches)) {
    $line = Preg::replace($pattern, ' ' . trim($matches[0]), $line);
}
```

but it does not pass the test (it's aligned, but operator is changed 😅):

```
 Caused by
 Code build on input code must match expected code.
 Failed asserting that two strings are identical.
 --- Expected
 +++ Actual
 @@ @@
  '<?php
  
 -             $testA = $testB["abc"]    / $testC * 100;
 +             $testA = $testB["abc"]    / $testC = 100;
               $testD = $testE["abcdef"] / $testF * 100;
              '
```

I don't want to dig deeper, maybe you already verified that path and can tell that it can't be done this way 😉. Or maybe I did some dumb mistake and it can work this way after polishing?

Reply:
@Wirone 

Thank you for your review.
Could you give me some time to think about it?

Reply:
Take any time you need, we all do it for free in our spare time 🙂.